### PR TITLE
Sortan filter for alpha-numeric filter

### PR DESCRIPTION
**Pull request by ibnishak**
Alpha numeric sorting. 

Demo of code in work [Demo](https://ibnishak.github.io/Tesseract/pluginsandmacros/index.html#Sortan%20-%20Sorting%20Alpha%20numerically)

Sorry for the formatting error

### CODE REVIEW

File: core/modules/filters/sort.js [hunk 1]
```diff
@@ -6,45 +6,57 @@ module-type: filteroperator
...
+	/*jslint node: true, browser: true */
+	/*global $tw: false */
+	"use strict";
+
+	/*
+	Export our filter function
+	*/
+	exports.sort = function(source, operator, options) {
+		var results = prepare_results(source);
+		options.wiki.sortTiddlers(results, operator.operand || "title", operator.prefix === "!", false, false);
```
Comment (pmario) [inline core/modules/filters/sort.js]: Hi, It seems you have changed core white-space formatting, which shouldn't be done. 
It should be like this:

```
options.wiki.sortTiddlers(results,operator.operand || "title",operator.prefix === "!",false,false);
```

no spaces after `,`

File: core/modules/wiki.js [hunk 2]
```diff
@@ -329,7 +329,7 @@ Sort an array of tiddler titles by a specified field
 	isDescending: true if the sort should be descending
 	isCaseSensitive: true if the sort should consider upper and lower case letters to be different
 */
-exports.sortTiddlers = function(titles,sortField,isDescending,isCaseSensitive,isNumeric) {
+exports.sortTiddlers = function(titles,sortField,isDescending,isCaseSensitive,isNumeric, isAlphaNumeric = false) {
```
Comment (pmario) [inline core/modules/wiki.js]: IMO this syntax causes browser compatibility problems. see: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters

File: core/modules/wiki.js [hunk 3]
```diff
@@ -358,9 +358,17 @@ exports.sortTiddlers = function(titles,sortField,isDescending,isCaseSensitive,is
...
+				return isDescending ? b.localeCompare(a, undefined, {numeric: true, sensitivity: 'variant'}) : a.localeCompare(b, undefined, {numeric: true, sensitivity: 'variant'});
+			}
+			else{
+				return isDescending ? b.localeCompare(a, undefined, {numeric: true, sensitivity: 'base'}) : a.localeCompare(b, undefined, {numeric: true, sensitivity: 'base'});
+
+			}
 		} else if($tw.utils.isDate(a) && $tw.utils.isDate(b)) {
 			return isDescending ? b - a : a - b;
-		} else {
+		}else {
```
Comment (pmario) [inline core/modules/wiki.js]: there should be a space in front of `else`

File: core/modules/wiki.js [hunk 4]
```diff
@@ -358,9 +358,17 @@ exports.sortTiddlers = function(titles,sortField,isDescending,isCaseSensitive,is
 		y = Number(b);
 		if(isNumeric && (!isNaN(x) || !isNaN(y))) {
 			return compareNumbers(x,y);
+		} else if(isAlphaNumeric) {
+			if (isCaseSensitive) {
+				return isDescending ? b.localeCompare(a, undefined, {numeric: true, sensitivity: 'variant'}) : a.localeCompare(b, undefined, {numeric: true, sensitivity: 'variant'});
+			}
+			else{
+				return isDescending ? b.localeCompare(a, undefined, {numeric: true, sensitivity: 'base'}) : a.localeCompare(b, undefined, {numeric: true, sensitivity: 'base'});
```
Comment (pmario) [inline core/modules/wiki.js]: it should be `} else {`

Comment (pmario) [reply core/modules/wiki.js]: whitespace after `,` ?!

**Comment by pmario** [comment]
>Sorry for the formatting error

uups, It seems you are aware already :)

File: core/modules/wiki.js (hunk 2 above)
Comment (pmario) [reply core/modules/wiki.js]: @Jermolene ??

**Comment by ibnishak** [comment]
Yes. Any chance to know which JS formatter is used for TW5 repo?

File: core/modules/wiki.js (hunk 2 above)
Comment (ibnishak) [reply core/modules/wiki.js]: Will check leaving it undefined

**Comment by pmario** [comment]
>Yes. Any chance to know which JS formatter is used for TW5 repo?

I don't think there is one. I'm using brackets editor with js-hint plugin. ... BUT I needed to tweak it, a bit

The problem is, Jeremy follows his rules. ... but ... sometimes readability / understanding the code is more important as the default rules :) ... 

So it's not 100% consistent.

**Comment by ibnishak** [comment]
All formatting in line.

**Comment by ibnishak** [comment]
Should I delete this PR and make a new one from a fresh branch so as not to pollute the commit history?

File: .gitignore [hunk 5]
```diff
@@ -3,3 +3,5 @@
 tmp/
 output/
 node_modules/
+
+editions/tw5.com/tiddlers/$__StoryList.tid
```
Comment (pmario) [inline .gitignore]: hmmm, not sure about this one. ... There are 6 "-server" editions and every one of them will create a $__StoryList.tid ... So we may create a separated PR to fix this behaviour. ...

I think we should be able to safely ignore every tiddler `$__StoryList.tid` for core editions. 

There is a problem if the DefaultTiddler is set to `[list[$:/StoryList]]` which prevented us from disabling the story list save at the TW saver level. ... But for core editions, DefaultTiddler isn't set.

**Comment by pmario** [comment]
>Should I delete this PR and make a new one from a fresh branch so as not to pollute the commit history?

Why. I think it's a good PR + some discussion and some fixes. ... That's the way it works. ... I don't see pollution atm. ... except .gitignore - which I think should be a separate PR

**Comment by ibnishak** [comment]
gitignore was not supposed to be in PR. I merely wanted to ignore the storylist in tw5.com edition. I assumed that gitignore won't be a part of commit - rookie mistake.

**Comment by Jermolene** [comment]
Thanks @ibnishak @pmario there are a couple of minor coding style issues, I'll fix them in subsequent commit

**Pull request merged**